src/geant4/app/src/ExN01DetectorConstruction.cc

```diff
@@ -0,0 +1,454 @@
+                                    1.053992E-3 * GeV, 1.108032E-3 * GeV, 1.164842E-3 * GeV, 1.194330E-3 * GeV, 1.224564E-3 * GeV,
+                                    1.287349E-3 * GeV, 1.353353E-3 * GeV, 1.422741E-3 * GeV, 1.495686E-3 * GeV, 1.533550E-3 * GeV,
+                                    1.572372E-3 * GeV, 1.612176E-3 * GeV, 1.652989E-3 * GeV, 1.737739E-3 * GeV, 1.826835E-3 * GeV,
+                                    1.873082E-3 * GeV, 1.920499E-3 * GeV, 1.969117E-3 * GeV, 2.018965E-3 * GeV, 2.122480E-3 * GeV,
+                                    2.231302E-3 * GeV, 2.268877E-3 * GeV, 2.306855E-3 * GeV, 2.345703E-3 * GeV, 2.365253E-3 * GeV,
+                                    2.385205E-3 * GeV, 2.425130E-3 * GeV, 2.465970E-3 * GeV, 2.592403E-3 * GeV, 2.725318E-3 * GeV,
+                                    2.865048E-3 * GeV, 3.011942E-3 * GeV, 3.088190E-3 * GeV, 3.166368E-3 * GeV, 3.246525E-3 * GeV,
+                                    3.328711E-3 * GeV, 3.499377E-3 * GeV, 3.678794E-3 * GeV, 3.867410E-3 * GeV, 4.065697E-3 * GeV,
+                                    4.274149E-3 * GeV, 4.493290E-3 * GeV, 4.607038E-3 * GeV, 4.723666E-3 * GeV, 4.843246E-3 * GeV,
+                                    4.965853E-3 * GeV, 5.220458E-3 * GeV, 5.488116E-3 * GeV, 5.769498E-3 * GeV, 5.915554E-3 * GeV,
+                                    6.050000E-3 * GeV, 6.218851E-3 * GeV, 6.376282E-3 * GeV, 6.537698E-3 * GeV, 6.592384E-3 * GeV,
+                                    6.647595E-3 * GeV, 6.703200E-3 * GeV, 6.872893E-3 * GeV, 7.046881E-3 * GeV, 7.225274E-3 * GeV,
+                                    7.408182E-3 * GeV, 7.595721E-3 * GeV, 7.788008E-3 * GeV, 7.985162E-3 * GeV, 8.187308E-3 * GeV,
+                                    8.394570E-3 * GeV, 8.607080E-3 * GeV, 8.824969E-3 * GeV, 9.048374E-3 * GeV, 9.277435E-3 * GeV,
+                                    9.512294E-3 * GeV, 9.753099E-3 * GeV, 1.000000E-2 * GeV, 1.025315E-2 * GeV, 1.051271E-2 * GeV,
+                                    1.077884E-2 * GeV, 1.105171E-2 * GeV, 1.133148E-2 * GeV, 1.161834E-2 * GeV, 1.191246E-2 * GeV,
+                                    1.221403E-2 * GeV, 1.252323E-2 * GeV, 1.284025E-2 * GeV, 1.316531E-2 * GeV, 1.349859E-2 * GeV,
+                                    1.384031E-2 * GeV, 1.419068E-2 * GeV, 1.454991E-2 * GeV, 1.491825E-2 * GeV, 1.529590E-2 * GeV,
+                                    1.568312E-2 * GeV, 1.608014E-2 * GeV, 1.648721E-2 * GeV, 1.690459E-2 * GeV, 1.733253E-2 * GeV,
+                                    1.777131E-2 * GeV, 1.822119E-2 * GeV, 1.868246E-2 * GeV, 1.915541E-2 * GeV, 1.964033E-2 * GeV,
+                                    2.000000E-2 * GeV
+                                   };
+
+  // G4 expects a vector
+  // but CI doesnt like using the C++11 std
+  std::vector<G4double> bin_bounds_v;
+  // hence this
+  bin_bounds_v.assign(bin_bounds, bin_bounds + 261);
```

Comment:
I would declare 261 a constant and use it in both places (here and in `bin_bounds` above), or use `size_of`.